server/appServices.js

```diff
@@ -85,6 +91,52 @@ function updateAppsAndWatch(namespace, kubeclient) {
   });
 }
 
+/**
+ * Watch the namespace for the Mobile Security Service.
+ * When the namespace is created then start watching changes
+ * to the MobileSecurityService custom resource
+ *
+ * @param {String} namespace - The managed namespace of the Mobile Security Service
+ * @param {*} kubeclient - The kubernetes API client
+ */
+function watchMobileSecurityService(namespace, kubeclient) {
+  const stream = kubeclient.api.v1.watch.namespace(namespace).getStream();
+  const jsonStream = new JSONStream();
+  stream.pipe(jsonStream);
+  jsonStream.on('data', event => {
+    if (event.type === 'ADDED') {
+      watchMobileSecurityServiceCR(namespace, kubeclient);
+    }
+  });
+}
+
+/**
+ * Watch for changes to the MobileSecurityService custom resource.
+ *
+ * @param {String} namespace - The managed namespace of the Mobile Security Service
+ * @param {*} kubeclient - The kubernetes API client
+ */
+function watchMobileSecurityServiceCR(namespace, kubeclient) {
+  const stream = kubeclient.apis[mobileSecurityServiceCRD.spec.group].v1alpha1.watch
+    .namespace(namespace)
+    .mobilesecurityservices()
+    .getStream();
+
+  const jsonStream = new JSONStream();
+  stream.pipe(jsonStream);
+
+  jsonStream.on('data', event => {
+    if (event.type === 'ADDED') {
+      DeviceSecurityService.disabled = false;
+    } else if (event.type === 'DELETED') {
+      DeviceSecurityService.disabled = true;
+    } else if (event.type !== 'MODIFIED') {
+      watchMobileSecurityServiceCR(namespace, kubeclient);
```

Comment:
@ziccardi - how familar are you with the `kubeclient`? For some reason, after a few times of installing and uninstalling the Mobile Security Service CR and namespace, then `event` becomes a string "/version", before hand it was an object.

This is the only way I have found to get it working, but I don't think it is a good approach as it is adding extra watches to the stream. Any ideas?